fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -1237,11 +1237,11 @@ public OlapTable selectiveCopy(Collection<String> reservedPartitions, boolean re
             return copied;
         }
 
-        Set<String> partNames = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
-        partNames.addAll(copied.getPartitionNames());
+        Set<String> reservedPartitionSet = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        reservedPartitionSet.addAll(reservedPartitions);
```

Comment:
```
CREATE TABLE `test_tablet` (
  `k1` date NULL COMMENT "",
  `k2` int(11) NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`k1`, `k2`)
PARTITION BY RANGE(`k1`)
(PARTITION p20210101 VALUES [('2021-01-01'), ('2021-01-02')),
PARTITION p20210102 VALUES [('2021-01-02'), ('2021-01-03')),
PARTITION P20210102 VALUES [('2021-01-03'), ('2021-01-04'))
)
DISTRIBUTED BY HASH(`k2`) BUCKETS 1
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
```
This create partition will failed because of partition name is case sensitive.
So why this place remove this restrictions?